modules/mapbox/src/mapbox-overlay.ts

```diff
@@ -212,9 +226,12 @@ export default class MapboxOverlay implements IControl {
 
   private _updateViewState = () => {
     const deck = this._deck;
-    if (deck) {
-      // @ts-ignore (2345) map is always defined if deck is
-      deck.setProps({viewState: getViewState(this._map)});
+    const map = this._map;
+    if (deck && map) {
```

Comment:
I think it is actually better to crash if our assumption is that `map` is defined is invalid